### PR TITLE
Test fixup for VSCode vs CLI behavior

### PR DESCRIPTION
**Pull request by bpkroth**

### CODE REVIEW

File: mlos_bench/mlos_bench/tests/launcher_parse_args_test.py [hunk 1]
```diff
@@ -70,7 +77,9 @@ def test_launcher_args_parse_1(config_paths: List[str]) -> None:
     # Check overriding values in a file from the command line.
     assert launcher.global_config['test_global_value_2'] == 'from-args'
     # Check that we can expand a $var in a config file that references an environment variable.
-    assert path_join(launcher.global_config["varWithEnvVarRef"], abs_path=True) == path_join(os.getcwd(), "foo", abs_path=True)
+    assert path_join(launcher.global_config["pathVarWithEnvVarRef"], abs_path=True) \
+        == path_join(expected_root_pwd, "foo", abs_path=True)
```
Comment (motus) [inline mlos_bench/mlos_bench/tests/launcher_parse_args_test.py:81]: maybe
```suggestion
    assert os.path.samepath(path_join(launcher.global_config["pathVarWithEnvVarRef"]),
                            path_join(expected_root_pwd, "foo"))
```

File: mlos_bench/mlos_bench/tests/launcher_parse_args_test.py [hunk 2]
```diff
@@ -105,7 +120,9 @@ def test_launcher_args_parse_2(config_paths: List[str]) -> None:
     # Check that secondary expansion also works.
     assert launcher.global_config['testVnetName'] == 'MockeryExperiment-vm-vnet'
     # Check that we can expand a $var in a config file that references an environment variable.
-    assert path_join(launcher.global_config["varWithEnvVarRef"], abs_path=True) == path_join(os.getcwd(), "foo", abs_path=True)
+    assert path_join(launcher.global_config["pathVarWithEnvVarRef"], abs_path=True) \
+        == path_join(expected_root_pwd, "foo", abs_path=True)
```
Comment (motus) [inline mlos_bench/mlos_bench/tests/launcher_parse_args_test.py:124]:
```suggestion
    assert os.path.samepath(path_join(launcher.global_config["pathVarWithEnvVarRef"]),
                            path_join(expected_root_pwd, "foo"))
```

**Review by motus — APPROVED** [review]
Looks good!

File: mlos_bench/mlos_bench/tests/launcher_parse_args_test.py (hunk 1 above)
Comment (bpkroth) [reply mlos_bench/mlos_bench/tests/launcher_parse_args_test.py:81]: The call is `samefile` I think, not `samepath` and doesn't work for files that don't exist unfortunately.

```python
>>> os.path.samefile("foo", "./foo")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<frozen genericpath>", line 100, in samefile
FileNotFoundError: [Errno 2] No such file or directory: 'foo'
```

**Pull request merged**